doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,62 +139,50 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
-                    :   [[|] `equation` | … | `equation`] end
-                    : `qualid`
-                    : `sort`
-                    : `num`
-                    : _
-                    : ( `term` )
-   arg              : `term`
-                    : ( `ident` := `term` )
-   binders          : `binder` … `binder`
-   binder           : `name`
-                    : ( `name` … `name` : `term` )
-                    : ( `name` [: `term`] := `term` )
-                    : ' `pattern`
-   name             : `ident` | _
-   qualid           : `ident` | `qualid` `field`
-   sort             : SProp | Prop | Set | Type
-   fix_bodies       : `fix_body`
-                    : `fix_body` with `fix_body` with … with `fix_body` for `ident`
-   cofix_bodies     : `cofix_body`
-                    : `cofix_body` with `cofix_body` with … with `cofix_body` for `ident`
-   fix_body         : `ident` `binders` [`annotation`] [: `term`] := `term`
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+.. insertgram term binders_opt
 
+.. productionlist:: coq
+   term : forall `open_binders` , `term`
+        : fun `open_binders` => `term`
+        : `term_let`
+        : if `term` `as_return_type_opt` then `term` else `term`
+        : `term_fix`
+        : `term100`
+   term100 : `term_cast`
+           : `term10`
+   term10 : `term1` `args`
+          : @ `qualid` `universe_annot_opt` `term1_list_opt`
+          : `term1`
+   args : `args` `arg`
+        : `arg`
+   arg : ( `ident` := `term` )
+       : `term1`
+   term1_list_opt : `term1_list_opt` `term1`
+                  : `empty`
+   empty :
+   term1 : `term_projection`
+         : `term0` % `ident`
+         : `term0`
+   args_opt : `args`
+            : `empty`
+   term0 : `qualid` `universe_annot_opt`
+         : `sort`
+         : `numeral`
+         : `string`
+         : _
+         : `term_evar`
+         : `term_match`
+         : ( `term` )
+         : {| `fields_def` |}
+         : `{ `term` }
+         : `( `term` )
+         : ltac : ( `ltac_expr` )
+   fields_def : `field_def` ; `fields_def`
+              : `field_def`
+              : `empty`
+   field_def : `qualid` `binders_opt` := `term`
+   binders_opt : `binders`
+               : `empty`
```

Comment:
I was suggesting to put this definition in the binders grammar, but I guess it is fine here too.